apps/crn-server/test/data-providers/contentful/calendars.data-provider.test.ts

```diff
@@ -148,6 +151,112 @@ describe('Calendars data provider', () => {
       });
     });
 
+    test('Should filter calendars by maxExpiration when it is greater than results expirationDate', async () => {
+      const contentfulGraphqlResponse = getContentfulCalendarsGraphqlResponse();
+      const responseExpirationDate =
+        contentfulGraphqlResponse.calendarsCollection?.items[0]
+          ?.googleApiMetadata.expirationDate;
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphqlResponse,
+      );
+
+      const result = await calendarDataProvider.fetch({
+        maxExpiration: responseExpirationDate + 1,
+      });
+
+      expect(contentfulGraphqlClientMock.request).toBeCalledWith(
+        expect.anything(),
+        {
+          limit: 50,
+          order: ['name_ASC'],
+          skip: 0,
+          where: {
+            googleApiMetadata_exists: true,
+          },
+        },
+      );
+      expect(result).toEqual({
+        total: 1,
+        items: [getContentfulCalendarDataObject()],
+      });
+    });
+
+    test('Should filter calendars by maxExpiration when it is less than results expirationDate', async () => {
+      const contentfulGraphqlResponse = getContentfulCalendarsGraphqlResponse();
+      const responseExpirationDate =
+        contentfulGraphqlResponse.calendarsCollection?.items[0]
+          ?.googleApiMetadata.expirationDate;
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphqlResponse,
+      );
+
+      const result = await calendarDataProvider.fetch({
+        maxExpiration: responseExpirationDate - 1,
+      });
+
+      expect(contentfulGraphqlClientMock.request).toBeCalledWith(
+        expect.anything(),
+        {
+          limit: 50,
+          order: ['name_ASC'],
+          skip: 0,
+          where: {
+            googleApiMetadata_exists: true,
+          },
+        },
+      );
+      expect(result).toEqual({
+        total: 0,
+        items: [],
+      });
+    });
+
+    test('Should filter calendars by resourceId and maxExpiration', async () => {
+      const modifiedContentfulGraphqlCalendar = getContentfulGraphqlCalendar();
+      modifiedContentfulGraphqlCalendar.googleApiMetadata!.resourceId! =
```

Comment:
yeah i think you should have at least 3 calendars returned from contentful, first with an expirationDate over the requested limit but the right resourceID, second with a non-matching resourceId but passing expirationDate and third with both conditions fulfilled.

Reply:
committed here: https://github.com/yldio/asap-hub/pull/3046/commits/5379062bd1f167df262232259814e0d67e295d56